src/core/client/admin/App/GlobalNotification/GlobalNotificationContext.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {
+  createContext,
+  ReactNode,
+  useContext,
+  useMemo,
+  useReducer,
+} from "react";
+
+interface State {
+  message: ReactNode | null;
+  visible: boolean;
+}
+
```

Comment:
had to create a global notification component using Context because the component has to appear at a higher level in the nesting order than routing would permit otherwise.